apps/geo/filter_set.py

```diff
@@ -16,3 +24,54 @@ def geo_area_label(self, queryset, name, value):
         if value:
             return queryset.filter(label__icontains=value)
         return queryset
+
+
+class RegionFilterSet(UserResourceFilterSet):
+    """
+    Region filter set
+
+    Filter by code, title and public fields
+    """
+    # NOTE: This filter the regions not in the supplied project
+    exclude_project = django_filters.ModelMultipleChoiceFilter(
+        method='exclude_project_region_filter',
+        widget=django_filters.widgets.CSVWidget,
+        queryset=Project.objects.all(),
+    )
+
+    class Meta:
+        model = Region
+        fields = ['id', 'code', 'title', 'public', 'project',
+                  'created_at', 'created_by', 'modified_at', 'modified_by']
+        filter_overrides = {
+            models.CharField: {
+                'filter_class': django_filters.CharFilter,
+                'extra': lambda f: {
+                    'lookup_expr': 'icontains',
+                },
+            },
+        }
+
+    def exclude_project_region_filter(self, qs, name, value):
+        if value:
+            return qs.filter(~models.Q(project__in=value)).distinct()
```

Comment:
This is fine but we can also use exclude here.
```suggestion
            return qs.exclude(project__in=value).distinct()
```